cadence/nfts/edition-nft/EditionNFT.template.cdc

```diff
@@ -236,22 +301,31 @@ pub contract {{ contractName }}: NonFungibleToken {
             let edition = {{ contractName }}.editions[editionID]
                 ?? panic("edition does not exist")
 
-            // Increase the edition count by one
-            edition.incrementCount()
+            // Increase the edition size by one
+            edition.incrementSize()
```

Comment:
Do we need a check / assert here to verify if the edition size has reached its limit? 

```
         if let limit = edition.limit {
                if edition.size == limit {
                    edition.close()
```

Wouldn't this line let the mint to go through?

Reply:
Thanks for pointing this out. I confirmed that the logic works correctly, but it's definitely confusing.

There was a precondition in `incrementSize` that prevented the caller from increasing the size of a closed edition. Given that an edition is always closed when `size == limit`, it was impossible to increase the size past the limit. However, I updated the code to hopefully make that enforcement more clear.

- I adjusted the test cases to specifically check this edge case (minting 1 more than the limit)
- I moved the `isClosed` check from the `Edition` struct into the `mintNFT` function. This way it's easier to see all the logic in one function.